### ISSUE TITLE
[no-unnecessary-condition] False positive - Value is always falsy

### ISSUE THREAD

**Issue by FlorinAsavoaie** [typescript-eslint/typescript-eslint#1176]
**Repro**

```JS
module.exports = {
  parser: "@typescript-eslint/parser",
  parserOptions: {
    ecmaVersion: 10,
    sourceType: "module",
    project: [
      "tsconfig.json"
    ],
  },
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  rules: {
    indent: ["error", 2],
    "@typescript-eslint/indent": ["error", 2],
    "@typescript-eslint/unified-signatures": "warn",
    "@typescript-eslint/quotes": "error",
    "@typescript-eslint/promise-function-async": "error",
    "@typescript-eslint/prefer-readonly": "error",
    "@typescript-eslint/prefer-for-of": "error",
    "@typescript-eslint/no-unnecessary-condition": "error",
    "@typescript-eslint/no-magic-numbers": "error",
    "@typescript-eslint/member-ordering": "error",
    "@typescript-eslint/member-naming": "error",
    "@typescript-eslint/brace-style": "error",
    "@typescript-eslint/array-type": "error"
  },
};
```

```TS
/* eslint-disable @typescript-eslint/no-unused-vars */
const myFunction = (someBool?: boolean): void => {
  const test = someBool || false;
};
```

**Expected Result**
No issue with code.

**Actual Result**
3:28  error  Unnecessary conditional, value is always falsy  @typescript-eslint/no-unnecessary-condition

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.12.0` |

**Comment by Retsam** [comment]
It's reporting an error because the `ignoreRhs` option isn't set, so it's treating the right hand side of the `||` operator as a conditional.  

Unfortunately, the current version of the rule requires you to choose between false-positives with `ignoreRhs` off, and some false-negatives with `ignoreRhs` on.  See #1017 .  Should be fixed by #1163 , but it's a breaking change that will need to wait for the next major version bump.

**Comment by bradzacher** [comment]
As this is working as expected - I'm going to close this.
